src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -0,0 +1,55 @@
+package ru.hogwarts.school.service;
+
+import org.springframework.stereotype.Service;
+import ru.hogwarts.school.exception.FacultyNotFoundException;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.repositories.FacultyRepository;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+
+@Service
+public class FacultyService {
+    private final FacultyRepository facultyRepository;
+
+    public FacultyService(FacultyRepository facultyRepository) {
+        this.facultyRepository = facultyRepository;
+    }
+
+
+    public Faculty createFaculty(Faculty faculty){
+        return facultyRepository.save(faculty);
+    }
+    public void updateFaculty(long id, Faculty faculty){
+        facultyRepository.findById(id).orElseThrow(
+                () -> new FacultyNotFoundException(id));
+        faculty.setId(id);
+        facultyRepository.save(faculty);
+    }
+    public Faculty getFaculty(long id){
+        return facultyRepository.findById(id).orElseThrow(
+                () -> new FacultyNotFoundException(id));
+    }
+    public Faculty removeFaculty(long id){
+        Faculty remoteFaculty = facultyRepository.findById(id).orElseThrow(
+                () -> new FacultyNotFoundException(id));
+
+        facultyRepository.deleteById(id);
```

Comment:
```suggestion
        facultyRepository.delete(remoteFaculty);
```
раз достал, то лучше так